lib/puppet/util/windows/sid.rb

```diff
@@ -75,7 +75,10 @@ def name_to_principal(name, allow_unresolved = false)
           raw_sid_bytes = sid_ptr.read_array_of_uchar(get_length_sid(sid_ptr))
         end
       rescue => e
-        Puppet.debug("Could not retrieve raw SID bytes from '#{name}': #{e.message}")
+        # Avoid debug logs pollution with valid account names
+        # https://docs.microsoft.com/en-us/windows/win32/debug/system-error-codes--1300-1699-
+        # https://docs.microsoft.com/en-us/windows/win32/api/sddl/nf-sddl-convertsidtostringsidw#return-value
+        Puppet.debug("Could not retrieve raw SID bytes from '#{name}': #{e.message}") unless e.code == 0x539
```

Comment:
Maybe reference the `ERROR_INVALID_SID_STRUCTURE` constant defined above?

Reply:
Is https://github.com/puppetlabs/puppet/pull/8652/files#diff-e04742a87f3cb0ad5f4c2fe71752b791b97894d34ba385d2d09d21375a9b3cc3R79 all right or did you have something else in mind?

Reply:
Ah sorry, I meant `unless e.code == ERROR_INVALID_SID_STRUCTURE`